lib/Frontend/nvqpp/ConvertExpr.cpp

```diff
@@ -604,18 +604,11 @@ Value QuakeBridgeVisitor::integerCoercion(Location loc,
                                           const clang::QualType &clangTy,
                                           Type dstTy, Value srcVal) {
   auto fromTy = getResultType(srcVal.getType());
-  if (dstTy == fromTy)
-    return srcVal;
-
   assert(fromTy.isa<IntegerType>() && dstTy.isa<IntegerType>());
-  if (fromTy.getIntOrFloatBitWidth() < dstTy.getIntOrFloatBitWidth()) {
-    auto mode = (clangTy->isUnsignedIntegerOrEnumerationType())
-                    ? cudaq::cc::CastOpMode::Unsigned
-                    : cudaq::cc::CastOpMode::Signed;
-    return builder.create<cudaq::cc::CastOp>(loc, dstTy, srcVal, mode);
-  }
-  assert(fromTy.getIntOrFloatBitWidth() > dstTy.getIntOrFloatBitWidth());
-  return builder.create<cudaq::cc::CastOp>(loc, dstTy, srcVal);
+  auto mode = (clangTy->isUnsignedIntegerOrEnumerationType())
```

Comment:
The mode should never be used when truncating a value.

Reply:
done, thanks!